### PR TITLE
STM32 :  add  STM32C071 series

### PR DESCRIPTION
**Pull request by djiatsaf-st**
- Add soc and kconfig
- Provide support dtsi files
- Add board support for nucleo_c071rb
- Run samples like helloworld, blinky and button

### CODE REVIEW

File: boards/st/nucleo_c071rb/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  - Mass storage
+  - Debug port
+
+More information about STM32C071RB can be found here:
+`STM32C0x1 reference manual`_
+
+Supported Features
+==================
+
+The Zephyr nucleo_c071rb board configuration supports the following hardware features:
```
Comment (nordicjm) [inline boards/st/nucleo_c071rb/doc/index.rst:58]:
```suggestion
The Zephyr ``nucleo_c071rb`` board supports the following hardware features:
```

File: boards/st/nucleo_c071rb/nucleo_c071rb_defconfig [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Serial Drivers
+CONFIG_SERIAL=y
+CONFIG_UART_INTERRUPT_DRIVEN=y
+
+# enable console
```
Comment (nordicjm) [inline boards/st/nucleo_c071rb/nucleo_c071rb_defconfig:7]:
```suggestion
# Enable console
```

File: boards/st/nucleo_c071rb/doc/img/nucleo_c071rb.jpg
Comment (nordicjm) [inline boards/st/nucleo_c071rb/doc/img/nucleo_c071rb.jpg:1]: Remove background (if possible), reduce width/height to remove black space, convert to webp, then put through https://tinypng.com/

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: boards/st/nucleo_c071rb/Kconfig.nucleo_c071rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+# Copyright (c) 2023 Benjamin Björnsson <benjamin.bjornsson@gmail.com>
```
Comment (FRASTM) [inline boards/st/nucleo_c071rb/Kconfig.nucleo_c071rb:1]: Copyright (c) 2024 etc

File: boards/st/nucleo_c071rb/arduino_r3_connector.dtsi [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+/ {
+	arduino_header: connector {
+		compatible = "arduino-header-r3";
+		#gpio-cells = <2>;
+		gpio-map-mask = <0xffffffff 0xffffffc0>;
+		gpio-map-pass-thru = <0 0x3f>;
+		gpio-map = <0 0 &gpioa 0 0>,	/* A0 */
+			   <1 0 &gpioa 1 0>,	/* A1 */
+			   <2 0 &gpioa 4 0>,	/* A2 */
+			   <3 0 &gpiob 1 0>,	/* A3 */
```
Comment (FRASTM) [inline boards/st/nucleo_c071rb/arduino_r3_connector.dtsi:16]: GPIO PB0 on A3, etc
see  the schematics

File: boards/st/nucleo_c071rb/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+Each of the GPIO pins can be configured by software as output (push-pull or open-drain), as
+input (with or without pull-up or pull-down), or as peripheral alternate function. Most of the
+GPIO pins are shared with digital or analog alternate functions. All GPIOs are high current
+capable except for analog inputs.
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- I2C1 SCL/SDA : PB8/PB9 (Arduino I2C)
+- LD4       : PA5
```
Comment (FRASTM) [inline boards/st/nucleo_c071rb/doc/index.rst:112]: LD1 and LD4 on PC9

**Review by FRASTM — CHANGES_REQUESTED** [review]

File: soc/st/stm32/stm32c0x/Kconfig.defconfig.stm32c071xx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+# STMicroelectronics STM32C071xx MCU
+
+# Copyright (c) STMicroelectronics.
```
Comment (erwango) [inline soc/st/stm32/stm32c0x/Kconfig.defconfig.stm32c071xx:3]:
```suggestion
# Copyright (c) 2024 STMicroelectronics.
```

File: soc/st/stm32/stm32c0x/Kconfig.soc [hunk 7]
```diff
@@ -10,6 +10,10 @@ config SOC_SERIES_STM32C0X
 config SOC_SERIES
 	default "stm32c0x" if SOC_SERIES_STM32C0X
 
+config SOC_STM32C071XX
+	bool
+	select SOC_SERIES_STM32C0X
```
Comment (erwango) [inline soc/st/stm32/stm32c0x/Kconfig.soc:15]: Please sort numerically

File: dts/arm/st/c0/stm32c071.dtsi [hunk 8]
```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023 Benjamin Björnsson <benjamin.bjornsson@gmail.com>
```
Comment (erwango) [inline dts/arm/st/c0/stm32c071.dtsi:2]: Copyright. Apply everywhere

File: boards/st/nucleo_c071rb/board.cmake [hunk 9]
```diff
@@ -0,0 +1,7 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# keep first
+board_runner_args(stm32cubeprogrammer "--port=swd" "--reset-mode=hw")
+
+# keep first
+include(${ZEPHYR_BASE}/boards/common/stm32cubeprogrammer.board.cmake)
```
Comment (erwango) [inline boards/st/nucleo_c071rb/board.cmake:7]: Please add openocd to enable debug

File: boards/st/nucleo_c071rb/nucleo_c071rb.dts [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+	vref-mv = <3300>;
+};
+
+&die_temp {
+	status = "okay";
+};
+
+&vref {
+	status = "okay";
+};
```
Comment (erwango) [inline boards/st/nucleo_c071rb/nucleo_c071rb.dts:146]: Did you test all those ?

**Review by erwango — CHANGES_REQUESTED** [review]
Otherwise LGTM

File: boards/st/nucleo_c071rb/nucleo_c071rb.dts (hunk 10 above)
Comment (djiatsaf-st) [reply boards/st/nucleo_c071rb/nucleo_c071rb.dts:146]: Yes, I did the tests on ADC and Dietemp

**Review by erwango — DISMISSED** [review]

**Review by FRASTM — DISMISSED** [review]

File: boards/st/nucleo_c071rb/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+  - Mass storage
+  - Debug port
+
+More information about STM32C071RB can be found here:
+`STM32C0x1 reference manual`_
+
+Supported Features
+==================
+
+The Zephyr ``nucleo_c071rb`` board configuration supports the following hardware features:
```
Comment (nordicjm) [inline boards/st/nucleo_c071rb/doc/index.rst:58]: -configuration

File: boards/st/nucleo_c071rb/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,163 @@
...
+
+
+For more details please refer to `STM32 Nucleo-64 board User Manual`_.
+
+Programming and Debugging
+*************************
+
+Nucleo C071RB board includes an ST-LINK/V2-1 embedded debug tool interface.
+
+Applications for the ``nucleo_c071rb`` board configuration can be built and
```
Comment (nordicjm) [inline boards/st/nucleo_c071rb/doc/index.rst:126]: -configuration

File: boards/st/nucleo_c071rb/nucleo_c071rb.dts [hunk 13]
```diff
@@ -0,0 +1,154 @@
...
+	pinctrl-0 = <&spi1_nss_pa4 &spi1_sck_pa5
+		     &spi1_miso_pa11 &spi1_mosi_pa12>;
+	pinctrl-names = "default";
+	status = "okay";
+};
+
+&adc1 {
+	pinctrl-0 = <&adc1_in0_pa0 &adc1_in1_pa1 &adc1_in4_pa4>;
+	pinctrl-names = "default";
+	st,adc-clock-source = <SYNC>;
```
Comment (gautierg-st) [inline boards/st/nucleo_c071rb/nucleo_c071rb.dts:134]:
```suggestion
	st,adc-clock-source = "SYNC";
```
Sorry for that, it's my fault

**Comment by erwango** [comment]
@nordicjm PTAL

**Review by nordicjm — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by FRASTM — APPROVED** [review]

**Pull request merged**